services/src/contexts/schema.sql

```diff
@@ -634,11 +634,123 @@ CREATE TABLE collection_children (
     PRIMARY KEY (parent, child)
 );
 
+CREATE TYPE "ArunaDataProviderDefinition" AS (
+    id uuid,
+    "name" text,
+    api_url text,
+    project_id text,
+    api_token text,
+    filter_label text,
+    cache_ttl int
+);
+
+CREATE TYPE "DatabaseConnectionConfig" AS (
+    host text,
+    port int,
+    "database" text,
+    schema text,
+    "user" text,
+    "password" text
+);
+
+CREATE TYPE "GbifDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioAbcdDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioCollectionsDataProviderDefinition" AS (
+    "name" text,
+    collection_api_url text,
+    collection_api_auth_token text,
+    abcd_db_config "DatabaseConnectionConfig",
+    pangaea_url text,
+    cache_ttl int
+);
+
+CREATE TYPE "TextMetaDataDefinitionKeyValue" AS (
+    key text,
+    value "MetaDataDefinition"
+);
+
+CREATE DOMAIN "LayerId" AS text;
+
+CREATE DOMAIN "LayerCollectionId" AS text;
+
+CREATE TYPE "MockLayer" AS (
+    id "LayerId",
+    "name" text,
+    "description" text,
+    symbology "Symbology",
+    provenance "Provenance",
+    workflow json, -- TODO: get rid of json type
+    properties "PropertyType" [],
+    metadata "TextTextKeyValue" []
+);
+
+CREATE TYPE "MockCollection" AS (
+    id "LayerCollectionId",
+    "name" text,
+    "description" text,
+    layers "MockLayer" []
+);
+
+CREATE TYPE "MockExternalLayerProviderDefinition" AS (
+    id uuid,
+    root_collection_id "LayerCollectionId",
+    root_collection_name text,
+    root_collection_description text,
+    root_collection_collections "MockCollection" [],
+    root_collection_layers "MockLayer" [],
+    "data" "TextMetaDataDefinitionKeyValue" []
```

Comment:
we store all the data of the whole mock provider in a single cell/row of a table, right? And every time the provider is loaded, all the data is loaded from Postgres into Geo Engine. This doesn't really scale to well.

Reply:
[afacdd0](https://github.com/geo-engine/geoengine/pull/866/commits/afacdd00f469216470891236b82bbec8bb532ea0)